instruments/cryomagnetics/cryomagnetics_4g.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+"""
+Implements the cryomagnetics 4G superconducting magnet power supply
+"""
+
+# IMPORTS #####################################################################
+from __future__ import absolute_import
+
+import re
+from time import sleep
+
+import quantities as pq
+
+from instruments.abstract_instruments.cryomagnetics_instrument \
+    import CryomagneticsInstrument
+
+# CLASSES #####################################################################
+
+
+class Cryomagnetics4G(CryomagneticsInstrument):
+    """
+    Implements a Cryomagnetics 4G Superconducting magnet power supply.
+    """
+    CHANNELS = {1, 2}
+
+    UNITS = {
+        "A": pq.amp,
+        "G": pq.gauss
+    }
+
+    REVERSE_UNITS = {
+        pq.amp: "A",
+        pq.gauss: "G"
+    }
+
+    instrument_measurement_wait = 0.5
+
+    @property
+    def unit(self):
+        """
+        The power supply is capable of expressing the output current going
+        into the magnet in amperes, but it can also convert the current to a
+        predicted magnetic field using a particular relation.
+
+        :return: The units in which the power supply expresses its measurement
+        """
+        response = self.query("UNITS?")
+        return self.UNITS[response]
+
+    @unit.setter
+    def unit(self, unit_to_set):
```

Comment:
A lof of these properties can probably be done with our property factories, then you'd get a bunch of extra protections that I've already programmed in. See https://github.com/Galvant/InstrumentKit/blob/master/instruments/util_fns.py